### PR TITLE
Prioritise current facility sync

### PR DESCRIPTION
**Pull request by ssrihari**

### CODE REVIEW

File: spec/controllers/shared_examples/shared_spec_for_sync_controller.rb [hunk 1]
```diff
@@ -211,6 +219,88 @@ def set_authentication_headers
   end
 end
 
+RSpec.shared_examples 'a working Current sync controller sending records' do
```
Comment (govindkrjoshi) [inline spec/controllers/shared_examples/shared_spec_for_sync_controller.rb]: Should we make separate files for V1 and current controller?

File: app/controllers/concerns/api/current/sync_to_user.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    end
+
+    def other_facilities_processed_since
+      process_token[:other_facilities_processed_since].try(:to_time) || Time.new(0)
+    end
+
+    def current_facility_processed_since
+      if process_token[:current_facility_processed_since].blank?
+        other_facilities_processed_since
+      elsif process_token[:current_facility_id] != current_facility.id
```
[the remark this answers is not in the document]
Comment (ssrihari) [reply app/controllers/concerns/api/current/sync_to_user.rb]: The client will mindlessly pass on the process token from one request to another. When the user moves from one facility to another, we'll run into this case, where the previous request (hence the process_token's current_facility_id) has a different facility id from the current request.

File: app/controllers/concerns/api/v1/sync_controller_overrides.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+module Api::V1::SyncControllerOverrides
+  extend ActiveSupport::Concern
+
+  included do
+    def model_name
```
[the remark this answers is not in the document]
Comment (ssrihari) [reply app/controllers/concerns/api/v1/sync_controller_overrides.rb]: Oh ya. Thanks!

File: spec/controllers/api/current/appointments_controller_spec.rb [hunk 4]
```diff
@@ -27,6 +27,36 @@
   end
 
   describe 'GET sync: send data from server to device;' do
-    it_behaves_like 'a working sync controller sending records'
+    it_behaves_like 'a working Current sync controller sending records'
+
+    describe 'current facility prioritisation' do
+      it "syncs request facility's records first" do
```
[the remark this answers is not in the document]
Comment (ssrihari) [reply spec/controllers/api/current/appointments_controller_spec.rb]: I know, I started off with a shared controller, actually. But the way it stands right now, it's not easy to extract. `registration_facility` vs `facility`, etc.

It'll be easy to extract when all the entities fall into the same pattern. Right now, many entities (communications, medical histories, etc) aren't prioritised by facility.

File: spec/controllers/shared_examples/shared_spec_for_sync_controller.rb (hunk 1 above)
Comment (ssrihari) [reply spec/controllers/shared_examples/shared_spec_for_sync_controller.rb]: Will extract when we have more shared specs for current. Right now, these specs have more things in common than not.

**Pull request merged**